compiler/rustc_hir_analysis/src/collect/item_bounds.rs

```diff
@@ -104,6 +104,11 @@ pub(super) fn explicit_item_bounds(
         None => {}
     }
 
+    // effects desugared associated types have no bounds.
```

Comment:
Could feed this query.

Also: don't we need a `where Self: Sized` bound to avoid breaking all `dyn Trait` for now?

Reply:
assoc types don't break object safety (we'd just need to make it default to `Runtime`)

what breaks object safety would probably be the `const host: bool` that we now insert onto each method.